scripts/run-clang-tidy.py

```diff
@@ -0,0 +1,628 @@
+#!/usr/bin/env python3
```

Comment:
Is the script truly necessary? Are there any alternatives? If so, what are they? This lengthy script looks like something I wouldn't be happy to maintain.

Reply:
Clang-tidy is quite slow because it needs to context of the entire compilation. 
This scripts is important to run the clang-tidy in concurrently based on your CPU cores.

Reply:
> Is the script truly necessary? Are there any alternatives? If so, what are they? This lengthy script looks like something I wouldn't be happy to maintain.

Actually you're right we don't need this file.
This file comes packaged with `clang-tidy` in the ubuntu package.
I've removed this file from the repo.

Reply:
fantastic, thank you!